src/library_fs.js

```diff
@@ -400,32 +400,44 @@ FS.staticInit();` +
     // SOCKFS is completed.
     createStream: (stream, fd_start, fd_end) => {
       if (!FS.FSStream) {
-        FS.FSStream = /** @constructor */ function() {
-          this.shared = { };
+        FS.FSStream = function FSStream() {
+          this.shared = {flags : 0, position : 0};
         };
-        FS.FSStream.prototype = {
+        FS.FSStream.prototype = Object.create(Object.prototype);
+        FS.FSStream.prototype.constructor = FS.FSStream;
+        Object.defineProperties(FS.FSStream.prototype, {
           object: {
-            get: function() { return this.node; },
-            set: function(val) { this.node = val; }
+            get: function () {
+              return this.node;
+            },
+            set: function (val) {
+              this.node = val;
+            },
+            enumerable : true,
```

Comment:
And, why do we need `enumerable` here? (Looks like we've never needed it before in our FS code.)